src/eko/anomalous_dimensions/as4/gNSm.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+"""
+This module contains the anomalous dimension :math:`\\gamma_{ns,-}^{(3)}`
+"""
+import numba as nb
+
+from ...constants import CF
+from ...harmonics.constants import zeta3
+
+
+@nb.njit(cache=True)
+def gamma_ns_nf3(n, sx):
+    """
+    Implements the common part proportional to :math:`nf^3`,
+    of :math:`\\gamma_{ns,+}^{(3)},\\gamma_{ns,-}^{(3)},\\gamma_{ns,v}^{(3)}`
+
+    The expression is copied exact from Eq. 3.6. of :cite:`Davies:2016jie`
+
+    Parameters
+    ----------
+        n : complex
+            Mellin moment
+        sx : list
+            harmonic sums cache
+
+    Returns
+    -------
+        g_ns_nf3 : complex
+            |N3LO| non-singlet anomalous dimension :math:`\\gamma_{ns}^{(3)}|_{nf^3}`
+    """
+    S1 = sx[0][0]
+    S2 = sx[1][0]
+    S3 = sx[2][0]
+    S4 = sx[3][0]
+    eta = 1 / n * 1 / (n + 1)
+    g_ns_nf3 = CF * (
+        -32 / 27 * zeta3 * eta
+        - 16 / 9 * zeta3
+        - 16 / 27 * eta**4
+        - 16 / 81 * eta**3
+        + 80 / 27 * eta**2
+        - 320 / 81 * eta
+        + 32 / 27 * 1 / (n + 1) ** 4
+        + 128 / 27 * 1 / (n + 1) ** 2
+        + 64 / 27 * S1 * zeta3
+        - 32 / 81 * S1
+        - 32 / 81 * S2
+        - 160 / 81 * S3
+        + 32 / 27 * S4
+        + 131 / 81
+    )
+    return g_ns_nf3
+
+
+@nb.njit(cache=True)
+def gamma_nsm_nf2(n, sx):
+    """
```

Comment:
I wonder if it is worth to repeat the references used function by function.

But maybe it is better like this, otherwise we bloat all docstrings with boilerplate. Just add a pointer (`see :mod:'X' for more details`) in order to make it easier to retrieve the source.